pkg/controller/catalogsourceconfig/configuring.go

```diff
@@ -95,26 +100,41 @@ func (r *configuringReconciler) createCatalogData(csc *v1alpha1.CatalogSourceCon
 
 // createCatalogSource creates a new CatalogSource CR and all the resources it
 // requires.
-func (r *configuringReconciler) createCatalogSource(csc *v1alpha1.CatalogSourceConfig) error {
+func (r *configuringReconciler) createCatalogSource(csc *v1alpha1.CatalogSourceConfig, client client.Client) error {
 	// Construct the operator artifact data to be placed in the ConfigMap data
 	// section.
 	data, err := r.createCatalogData(csc)
 	if err != nil {
 		return err
 	}
 
+	ensureGVK(csc)
```

Comment:
It's needed because the latest version of the sdk doesn't fill in the group version and kind of any of the objects that it returns from GET requests in the client. When I asked the SDK team about it they told me that is expected and that we should just manually fill those fields since the metadata is well known to us.

Reply:
Please add a comment mentioning this.

Reply:
Comments added